### ISSUE TITLE
How to use typings?

### ISSUE THREAD

**Issue by delagen** [jmdobry/CacheFactory#24]
```
import CacheFactory = require("cachefactory");
const base: CacheFactory.Cache = new CacheFactory().createCache("cache", {});
```

TS2503:Cannot find namespace 'CacheFactory'.

**Comment by delagen** [comment]
I think is must be written in another way. See lodash typings. Their typings are the best I ever seen

**Comment by jmdobry** [comment]
Can you please include in this issue the steps to reproduce the error you're seeing? The chances I'll debug for you go up exponentially when you do.

**Comment by delagen** [comment]
tsconfig.json

``` json
{
    "files": [
        "node_modules/cachefactory/dist/cachefactory.d.ts",
        "test.ts"
    ],
    "compilerOptions": {
        "module": "commonjs",
        "moduleResolution": "node",
        "lib": [
            "es6",
            "dom"
        ],
        "target": "es5",
        "noImplicitAny": true
    }
}
```

test.ts

``` ts
// without next line or entry in files of tsconfig.json this TS2307:Cannot find module 'cachefactory'.
/// <reference path="./node_modules/cachefactory/dist/cachefactory.d.ts" />
// with require import
// TS2503:Cannot find namespace 'CacheFactory'.
// TS2339:Property 'createCache' does not exist on type 'typeof CacheFactory'.
// import CacheFactory = require("cachefactory");
import * as CacheFactory from "cachefactory";

const cache:CacheFactory.Cache = CacheFactory.createCache("a", {});

console.log(CacheFactory, cache);
```

at least this 
TS2497:Module ''cachefactory'' resolves to a non-module entity and cannot be imported using this construct.

Node: 6.7.0
Typescript: 2.0.3

**Comment by fracz** [comment]
I can confirm the typings do not work.

Attempt: `import CacheFactory from 'cachefactory';`
Result: `TS1192: Module ''cachefactory'' has no default export.` 

Attempt: `import * as CacheFactory from "cachefactory"; new CacheFactory();`
Result: `TS2350: Only a void function can be called with the 'new' keyword.` 

Attempt: `import * as CacheFactory from "cachefactory"; CacheFactory.createCache("test", {});`
Result: `TS1316: Global module exports may only appear at top level.` 

Attempt: `import {CacheFactory} from "cachefactory";`
Result: `TS2305: Module ''cachefactory'' has no exported member 'CacheFactory'.` 

TS v2.1.4

**Comment by delagen** [comment]
PR #28

**Comment by delagen** [comment]
Any news in merge #29 as supersede

**Comment by jmdobry** [comment]
Fixed by #29
